samples/zip-split-rabbit-binder/src/test/java/com/example/ZipSplitRabbitBinderApplicationTests.java

```diff
@@ -0,0 +1,72 @@
+package com.example;
+
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.RabbitMQContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import org.springframework.amqp.core.ExchangeTypes;
+import org.springframework.amqp.rabbit.annotation.Exchange;
+import org.springframework.amqp.rabbit.annotation.Queue;
+import org.springframework.amqp.rabbit.annotation.QueueBinding;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Import(ZipSplitRabbitBinderApplicationTests.TestConfiguration.class)
+@SpringBootTest
+@DirtiesContext
+class ZipSplitRabbitBinderApplicationTests {
+
+	static Log LOG = LogFactory.getLog(ZipSplitRabbitBinderApplicationTests.class);
+
+	static BlockingQueue<String> DATA_SINK = new LinkedBlockingQueue<>();
+
+	@DynamicPropertySource
+	static void testProperties(DynamicPropertyRegistry registry) {
+		registry.add("file.supplier.directory", () -> new ClassPathResource("/dirWithZips").getPath());
+	}
+
+	@Test
+	void zippedFilesAreSplittedToRabbitBinding() throws InterruptedException {
+		List<String> expected = List.of("data111", "data112", "data113", "data121", "data122", "data123", "data124",
+				"data211", "data212", "data221", "data231", "data232", "data233", "data234", "data235");
+
+		for (int i = 0; i < expected.size(); i++) {
+			assertThat(DATA_SINK.poll(10, TimeUnit.SECONDS)).isIn(expected);
+		}
+	}
+
+	@org.springframework.boot.test.context.TestConfiguration(proxyBeanMethods = false)
+	static class TestConfiguration {
+
+		@Bean
```

Comment:
Any particular reason not to pull this up to a member of `ZipSplitRabbitBinderApplicationTests ` such as:
```java
@Container
@ServiceConnection
static RabbitMQContainer rabbitContainer = new RabbitMQContainer(DockerImageName.parse("rabbitmq:latest"));
```
I think this is the more common approach and simplifies the test a bit IMO. Wdyt?

Reply:
This is a result of `start.spring.io` when I chose `spring-boot-testcontainers`.
And that one generates for us an extra `TestConfiguration` class with such a bean.
I'm moving it as you are suggesting.
Looks like indeed makes test class looking much shorter.
However for that `@ServiceConnection` we still have to keep `spring-boot-testcontainers` 😄